### DISCUSSION TITLE
Allowing unused template literals

### DISCUSSION THREAD

**Discussion by colinhacks** [biomejs/biome#2949] [Ideas]
I'm just being annoying at this point, but I don't think it's necessarily widely believed that "unused template literals" are a bad practice. I'm very open to evidence to the contrary!

The proposal here is two-fold:

- do not enable `noUnusedTemplateLiterals` by default
- support `javascript.formatter.quoteStyle: "backtick"`

Some related Twitter discussion: https://twitter.com/colinhacks/status/1793115510672134326

**Comment by ematipico** [comment]
The first proposal sounds reasonable, however it would be a breaking change for our versioning contract, because it's a recommended rule, and removing a recommended rule requires a major.

The second proposal can't be implemented, unfortunately. Template literals can't be used as replacement for ALL string literals. If we can't replace all string literals, which quote do we use? For example, it can't be used inside module assertions. The following snippet won't parse:

```js
import {names} from "module" with { source: `json` } 
```

**Reply by colinhacks** [reply]
> The first proposal sounds reasonable, however it would be a breaking change for our versioning contract, because it's a recommended rule, and removing a recommended rule requires a major.

Of course, make sense 🫶

> The second proposal can't be implemented, unfortunately.

Makes sense. Certainly import statements, module assertions, JSX, etc are special cases. Are you opposed to any proposal where these situations are special cased in the formatter? I have ideas for how this can be handled in the config, but if this is a non-starter I won't waste your time.

**Reply by Conaclos** [reply]
We could add a new option `templateString: true/false` that enforces the usage of backticks for strings.

**Reply by colinhacks** [reply]
Hi Victorien!

That was one of my proposals, though it's not immediately clear to a user how that interacts with `quoteStyle` or which one takes precedence.

Here's another option:

```
javascript.formatter.quoteStyle: "backtick+single"
javascript.formatter.quoteStyle: "backtick+double"
```

I think this does a pretty good job of conveying a preference for backticks, while also requiring the user to specify a "fallback" for situations where backticks aren't supported.  It also keeps everything under one option and eliminates some potential for ambiguity.
